pallets/liquid-staking-v2/src/types.rs

```diff
@@ -0,0 +1,131 @@
+use codec::{Decode, Encode, FullCodec};
+use sp_runtime::{
+    traits::{AtLeast32BitUnsigned, MaybeSerializeDeserialize},
+    FixedPointNumber, FixedPointOperand, RuntimeDebug,
+};
+use sp_std::fmt::Debug;
+
+use primitives::Rate;
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum StakingOperationType {
+    Bond,
+    Unbond,
+    Rebond,
+    Matching,
+    TransferToRelaychain,
+    RecordRewards,
+    RecordSlashes,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum ResponseStatus {
+    Pending,
+    Processing,
+    Succeeded,
+    Failed,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub struct Operation<BlockNumber, Balance> {
+    pub amount: Balance,
+    pub block_number: BlockNumber,
+    pub status: ResponseStatus,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Default, Encode, Decode, RuntimeDebug)]
+pub struct PoolLedger<Balance> {
+    pub total_unstake_amount: Balance,
+    pub total_stake_amount: Balance,
+    pub operation_type: Option<StakingOperationType>,
+}
+
+impl<Balance> PoolLedger<Balance>
+where
+    Balance: AtLeast32BitUnsigned + FullCodec + Copy + MaybeSerializeDeserialize + Debug + Default,
+{
+    pub fn todo_after_new_era(&self) -> (StakingOperationType, Balance) {
```

Comment:
actually I suggest this should be:

`todo_before_new_era` so that if stake-client died for some eras, it can call first todo_before_new_era, if it succeeds then it can update the era if not it shouldn't